clusterctl/clusterdeployer/clusterdeployer.go

```diff
@@ -440,3 +535,10 @@ func containsMasterRole(roles []clustercommon.MachineRole) bool {
 	}
 	return false
 }
+
+func closeClient(client ClusterClient, name string) {
```

Comment:
Let this return `error` and let the caller decide what to do with it, if anything.